src/Handlers/PaymentResponseHandlerResult.php

```diff
@@ -29,7 +31,7 @@ public function createFromPaymentResponse($paymentResponse)
         $response = json_decode($response, true);
 
         if (json_last_error() !== JSON_ERROR_NONE) {
-            //TODO error handling
+            throw new PaymentFailedException('Invalid payment response data');
```

Comment:
I don't think so in this case, the payment response comes from Adyen API and we are [storing it](https://github.com/Adyen/adyen-shopware6/blob/4988b74e5a1d8bc07a15be269a874dc359dbbc2f/src/Handlers/PaymentResponseHandler.php#L135), so if it is invalid JSON, then it would be a server error and we need to talk to Checkout

Reply:
true! +1